integration/loki_micro_services_delete_test.go

```diff
@@ -115,4 +117,48 @@ func TestMicroServicesDeleteRequest(t *testing.T) {
 		require.Equal(t, `{job="fake"} |= "lineB"`, deleteRequests[0].Query)
 		require.Equal(t, "received", deleteRequests[0].Status)
 	})
+
+	// Wait until delete request is finished
+	t.Run("wait-until-delete-request-processed", func(t *testing.T) {
+		require.Eventually(t, func() bool {
+			deleteRequests, err := cliCompactor.GetDeleteRequests()
+			require.NoError(t, err)
+			require.Len(t, deleteRequests, 1)
+			return deleteRequests[0].Status == "processed"
+		}, 10*time.Second, 1*time.Second)
+
+		// Check metrics
+		metrics, err := cliCompactor.Metrics()
+		require.NoError(t, err)
+		checkLabelValue(t, "loki_compactor_delete_requests_processed_total", metrics, tenantID, 1)
+		// Re-enable this once flush works
+		// checkLabelValue(t, "loki_compactor_deleted_lines", metrics, tenantID, 1)
+	})
+
+	// Query lines, lineB should not be there
+	t.Run("query", func(t *testing.T) {
+		resp, err := cliQueryFrontend.RunRangeQuery(`{job="fake"}`)
+		require.NoError(t, err)
+		assert.Equal(t, "streams", resp.Data.ResultType)
+
+		var lines []string
+		for _, stream := range resp.Data.Stream {
+			for _, val := range stream.Values {
+				lines = append(lines, val[1])
+			}
+		}
+		// Remove lineB once flush works
+		assert.ElementsMatch(t, []string{"lineA", "lineB", "lineC", "lineD"}, lines)
```

Comment:
when we get to the point that `lineB` is actually removed, does `ElementsMatch` accept a 4th argument for a "reason" or "comment" string? If so I'd like to move the comment on 138 into there so that's printed out for the user when the test fails WDYT?

Reply:
Good idea. It has been added.